src/blog/2024/01/speech-driven-chatbot-with-node-red.md

```diff
@@ -0,0 +1,229 @@
+authors: ["sumit-shinde"]
+image: /blog/2024/01/images/speech-driven-chatbot-system.gif
+tags:
+    - posts
+    - node-red
+    - dashboard
+---
+
+Have you ever wondered about the complexity of integrating speech recognition and synthesis into your Node-RED projects? Often it requires external services or APIs. Today with this guide, I will show you how we can use speech recognition and synthesis in your Node-RED projects without needing an external service or API.  
+
+Not only this, I will make this guide more interesting by building a system that can listen to us and respond like humans using the Chat-GPT API.
+Let's get started!
+
+<!--more-->
+
+# What exactly is speech recognition and synthesis?
+
+Speech recognition is a technology where our device captures spoken words through a microphone, checks against grammar rules and vocabulary, and returns recognized words as text. On the other hand, Speech synthesis converts app text into speech and plays it through a device's speaker or audio output. There are a bunch of benefits and real-world applications of this technology.
+
+- **Hands-Free Operation:** Nowadays using speech recognition technology we can perform tasks such as making calls, sending messages, or controlling smart home devices without the need for physical interaction.
+- **Accessibility:** It allows individuals with visual impairments to access digital content through spoken words and as discussed above, to control devices without physical interaction.
+- **Efficient Content Consumption:** It allows us to listen to information instead of reading. For example, in the audiobook industry by using speech synthesis technology they create audio versions of books which helps users to be more productive.
+
+# Installing Dashboard 2.0 
+
+Now we have enough knowledge about speech recognition and synthesis technology and its real-world applications. Buckle up and Get ready for practical, so let's first start by installing Dashboard 2.0. 
+
+1. Install Node-RED Dashboard 2.0. Follow these [instructions](https://dashboard.flowfuse.com/getting-started.html) to install.
+2. Create your first group, page, theme, and base.
+
+# Building Speech-to-Text Vue component 
+In this section, we will build a Vue component that will perform a speech-to-text conversion operation using Web speech API for us and display results on the dashboard. Now you might say “You told us that we will not need any external API for speech recognition, right!”. Yes, you read it correctly! The [Web Speech API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Speech_API)
+is not any external API and it will process your speech locally, it is a JavaScript API that allows us to use speech-related functionalities, such as speech recognition and synthesis, in a web browser directly. It is widely present in modern browsers except firefox, which eliminates the need for external APIs to implement these features. That's quite a bit, now let's start to build that component.
```

Comment:
```suggestion
This will process your speech locally as it is a JavaScript API that allows us to use speech-related functionalities, such as speech recognition and synthesis, in a web browser directly. It is widely present in modern browsers (except Firefox) which eliminates the need for external APIs to implement these features. Let's now start to build that component.
```